vite.config.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference types="vitest" />
+import { fileURLToPath } from 'node:url'
+import { defineConfig } from 'vite'
+export default defineConfig({
+  define: {
+    __DEV__: true,
+    __TEST__: true,
+    __BROWSER__: true,
+  },
+  resolve: {
+    alias: [
+      {
+        find: /^@pinia\/(.*?)$/,
+        replacement: fileURLToPath(
+          new URL('./packages/packages/$1/src', import.meta.url)
+        ),
+      },
+      {
+        find: /^pinia$/,
+        replacement: fileURLToPath(
+          new URL('./packages/pinia/src', import.meta.url)
+        ),
+      },
+    ],
+  },
+  test: {
+    coverage: {
+      reporter: ['html', 'lcov', 'text'],
+      include: ['packages/pinia/src/**/*.ts', 'packages/testing/src/**/*.ts'],
+      exclude: [
+        '**/src/index.ts',
+        '**/*.d.ts',
+        '**/src/devtools',
+        '**/src/hmr.ts',
+        '**/src/deprecated.ts',
+        '**/src/vue2-plugin.ts',
+      ],
+    },
+    environment: 'happy-dom',
+    globals: true, // Specifically to make createTestingPinia happy
```

Comment:
what do you mean? 😮

Reply:
by default vitest doesn't expose global objects as jest does, so `vi` is not defined by default. We could either update the createTestingPinia tests to pass vi.fn. Or we can use this globals option.

Reply:
as long as the type of the variable `createSpy()` isn't `any` in the source code 😆